src/axom/core/memory_management.hpp

```diff
@@ -22,6 +22,13 @@
 namespace axom
 {
 
+#ifdef AXOM_USE_UMPIRE
+const int DEFAULT_ALLOCATOR_ID =
+  umpire::ResourceManager::getInstance().getAllocator("HOST").getId();
+#else
+constexpr int DEFAULT_ALLOCATOR_ID = 0;
+#endif
+
```

Comment:
I see this is in a header, but do you have to worry about when this is initialized? 
Could this cause a static initialization issue in the future?